src/interface/variable.hpp

```diff
@@ -204,6 +204,40 @@ class EdgeVariable {
   std::string label_;
 };
 
+template <typename T>
+class ParticleVariable {
+ public:
+  /// Initialize a particle variable
+  ParticleVariable(const std::string label, const int npool, const Metadata &metadata)
+      : data(label, npool), npool_(npool), m_(metadata), label_(label) {}
+
+  // accessors
+  KOKKOS_FORCEINLINE_FUNCTION
+  ParArrayND<T> &Get() { return data; }
+  template <typename... Args>
+  KOKKOS_FORCEINLINE_FUNCTION T &operator()(Args... args) { // const {
+    return data(std::forward<Args>(args)...);
+  }
+
+  ///< retrieve metadata for variable
+  const Metadata metadata() const { return m_; }
```

Comment:
You should be able to return this as a reference. Make the method noexcept.

Reply:
Are we fully confident that that this being a reference is fine? I'm worried that could turn into a memory error down the line, and `Metadata` is very lightweight, right?

Reply:
That's a fair point, use your better judgement. I might have been a bit over zealous in suggesting returning references.

Reply:
Roger that, thanks for bringing this up, it's a good thing to keep in mind.